daemon/client/conn/conn.go

```diff
@@ -101,25 +102,37 @@ func (c *conn) Echo(arg string) (string, error) {
 	return r.Pong, nil
 }
 
-func (c *conn) Run(cmd *runner.Command) (runner.ProcessStatus, error) {
+func (c *conn) Run(cmd *runner.Command) runner.ProcessStatus {
 	req := &protocol.Command{}
 	req.Argv = cmd.Argv
 	req.Env = cmd.EnvVars
 	req.Timeout = int64(cmd.Timeout)
 
 	r, err := c.client.Run(context.Background(), req)
 	if err != nil {
-		return runner.ProcessStatus{}, err
+		return runner.ProcessStatus{State: runner.FAILED, Error: err.Error()}
 	}
-	return protocol.ToRunnerStatus(r), nil
+	return protocol.ToRunnerStatus(r)
 }
 
-func (c *conn) Status(run runner.RunId) (runner.ProcessStatus, error) {
+func (c *conn) Status(run runner.RunId) runner.ProcessStatus {
 	r, err := c.client.Status(context.Background(), &protocol.StatusQuery{RunId: string(run)})
 	if err != nil {
-		return runner.ProcessStatus{}, err
+		return runner.ProcessStatus{State: runner.FAILED, Error: err.Error()}
 	}
-	return protocol.ToRunnerStatus(r), nil
+	return protocol.ToRunnerStatus(r)
+}
+
+func (c *conn) StatusAll() []runner.ProcessStatus {
```

Comment:
I agree I'm not sure the best way to proceed here, but Im a fan of starting with the simplest thing that works, so the current idea of this just being another implementation of Runner works for now.  And then we can iterate if this becomes problematic.  

It also makes it really clear in code that this is the equivalent to the Runner in the CloudWorker.  Which is a nice parallel to draw.